### PR TITLE
BUG:Clip with a list-like threshold with a nan is broken (GH19992)

### PR DESCRIPTION
**Pull request by makbigc**
- fix bug #19992
- 2 tests amended in frame/test_analytics.py and series/test_analytics.py
- whatsnew entry added

### CODE REVIEW

File: doc/source/whatsnew/v0.24.0.txt [hunk 1]
```diff
@@ -440,6 +440,7 @@ Missing
 
 - Bug in :func:`DataFrame.fillna` where a ``ValueError`` would raise when one column contained a ``datetime64[ns, tz]`` dtype (:issue:`15522`)
 - Bug in :func:`Series.hasnans` that could be incorrectly cached and return incorrect answers if null elements are introduced after an initial call (:issue:`19700`)
+- Bug in :func:`Series.clip` and `DataFrame.clip` cannot accept list-like threshold containing nan (:issue:`19992`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.24.0.txt]: can you add the ref for DataFrame, use double back-ticks on ``NaN`` (and sp)

File: pandas/tests/frame/test_analytics.py [hunk 2]
```diff
@@ -2237,13 +2237,20 @@ def test_clip_with_na_args(self):
...
         # GH # 17276
         tm.assert_frame_equal(self.frame.clip(np.nan), self.frame)
-        tm.assert_frame_equal(self.frame.clip(upper=[1, 2, np.nan]),
-                              self.frame)
-        tm.assert_frame_equal(self.frame.clip(lower=[1, np.nan, 3]),
-                              self.frame)
         tm.assert_frame_equal(self.frame.clip(upper=np.nan, lower=np.nan),
                               self.frame)
 
+        # GH #19992
```
Comment (jreback) [inline pandas/tests/frame/test_analytics.py]: can you comment under the commit number what this is testing

use 
```
result = 
expected = 
tm.assert_frame_equal(....)
```

do the tests in order for axis=0, axis=0, IOW so you don't have the df1, df2 variables

**Review by jreback — CHANGES_REQUESTED** [review]
small comments. lgtm otherwise. ping on green.

**Comment by jreback** [comment]
does this also close #19976 ?

File: pandas/tests/frame/test_analytics.py (hunk 2 above)
Comment (makbigc) [reply pandas/tests/frame/test_analytics.py]: Amended. Your format is better. Thanks.

File: doc/source/whatsnew/v0.24.0.txt [hunk 3]
```diff
@@ -440,6 +440,7 @@ Missing
 
 - Bug in :func:`DataFrame.fillna` where a ``ValueError`` would raise when one column contained a ``datetime64[ns, tz]`` dtype (:issue:`15522`)
 - Bug in :func:`Series.hasnans` that could be incorrectly cached and return incorrect answers if null elements are introduced after an initial call (:issue:`19700`)
+- Bug in :func:`Series.clip` and :func:`DataFrame.clip` cannot accept list-like threshold containing ``NaN`` (:issue:`19992`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.24.0.txt]: go ahead and move to 0.23.4

**Review by jreback — CHANGES_REQUESTED** [review]
does this close #19976, pls move to 0.23.4 whatnew

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @makbigc

**Pull request merged**